web/src/message_list_data.ts

```diff
@@ -260,32 +274,36 @@ export class MessageListData {
         });
     }
 
-    unmuted_messages(messages) {
+    unmuted_messages(messages: Message[]): Message[] {
         return this.messages_filtered_for_topic_mutes(
             this.messages_filtered_for_user_mutes(messages),
         );
     }
 
-    update_items_for_muting() {
+    update_items_for_muting(): void {
         this._items = this.unmuted_messages(this._all_items);
     }
 
-    first_unread_message_id() {
+    first_unread_message_id(): number | undefined {
         const first_unread = this._items.find((message) => unread.message_unread(message));
 
         if (first_unread) {
             return first_unread.id;
         }
 
         // if no unread, return the bottom message
-        return this.last().id;
+        return this.last()?.id;
```

Comment:
This is a behavior change; this function would previously throw an error rather than returning `undefined`. Returning `undefined` seems better, but we need to adjust all its callers to handle `undefined`.

Reply:
I tried adding a commit that does this, but I'm not actually sure if it's better since all the callers probably want to assert that it's defined anyways. Let me know what you think

Reply:
I think moving the `assert` to the caller makes sense since the reason why it should succeed is caller-specific.

But do use `assert(id !== undefined)`, not `assert(id)`, to avoid confusing `undefined` with `0`.

Reply:
Added `undefined` specific checks in a bunch of places